### PR TITLE
Updated RELEASES.md for 1.38.0

### PR DESCRIPTION
**Pull request by XAMPPRocky**
### [Rendered](https://github.com/XAMPPRocky/rust/blob/master/RELEASES.md)

r? @Mark-Simulacrum 
cc @rust-lang/release

### CODE REVIEW

**Comment by Centril** [comment]
What a boring release. 😂 

Missing?:

- https://github.com/rust-lang/rust/pull/60066
- https://github.com/rust-lang/rust/pull/60260

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,109 @@
...
+- [`ascii::EscapeDefault` now implements `Clone` and `Display`.][63421]
+- [Derive macros for prelude traits (e.g. `Clone`, `Debug`, `Hash`) are now
+  available at the same path as the trait.][63056] e.g. The `Clone` derive macro
+  is available at `std::clone::Clone`.
+- [`str::Chars` now implements `Debug`.][63000]
+- [`slice::{concat, connect, join}` now accepts `&[T]` in addition to `&T`.][62528]
+- [`*const T` and `*mut T` now implement `marker::Unpin`.][62583]
+- [`Arc<[T]>` and `Rc<[T]>` now implement `FromIterator<T>`.][61953]
+- [Built-in macros (e.g. `include!`, `line!`) are now available through the
+  `std` and `core` prelude.][62086] e.g. `core::prelude::v1::include`.
```
Comment (petrochenkov) [inline RELEASES.md]: I'd omit this item.
It's more important to mention that built-in macros are now available through std and core root like `core::include!(...)`, which is a part of the same PR as "Derive macros for prelude traits..." above (so I'd put it closely to that item).

Comment (XAMPPRocky) [reply RELEASES.md]: Okay, I've removed this item and reworded the earlier PR to include what you mentioned.

**Comment by XAMPPRocky** [comment]
@Centril Added those PRs.

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,113 @@
...
+- [`Duration::as_secs_f64`]
+- [`Duration::div_duration_f32`]
+- [`Duration::div_duration_f64`]
+- [`Duration::div_f32`]
+- [`Duration::div_f64`]
+- [`Duration::from_secs_f32`]
+- [`Duration::from_secs_f64`]
+- [`Duration::mul_f32`]
+- [`Duration::mul_f64`]
+- [`intrinsics::type_name`]
```
Comment (Centril) [inline RELEASES.md]:
```suggestion
- [`any::type_name`]
```

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,113 @@
...
+- [The `#[global_allocator]` attribute can now be used in submodules.][62735]
+- [The `#[deprecated]` attribute can now be used on macros.][62042]
+
+Compiler
+--------
+- [Added support for the `aarch64-uwp-windows-msvc`, `i686-uwp-windows-gnu`,
+  `i686-uwp-windows-msvc`, `x86_64-uwp-windows-gnu`, and
+  `x86_64-uwp-windows-msvc` targets.][60260]
+- [Added support for the `armv7-unknown-linux-gnueabi` and
+  `armv7-unknown-linux-musleabi` targets.][63107]
```
Comment (Mark-Simulacrum) [inline RELEASES.md]: We should note that these are tier 3 (I believe, not sure) targets here to give appropriate expectations for end-users.

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,113 @@
...
+Cargo
+-----
+- [Added pipelined compilation support to `cargo`.][cargo/7143]
+- [You can now pass the `--features` option multiple times to enable
+  multiple features.][cargo/7084]
+
+Misc
+----
+- [`rustc` will now warn if you create an invalid value, such as a null
+  reference with `mem::{uninitialized, zeroed}`.][63346]
```
Comment (Mark-Simulacrum) [inline RELEASES.md]: Probably some more loose language is warranted here, e.g. "for some invalid values" or something like that.

Comment (Mark-Simulacrum) [reply RELEASES.md]: cc @RalfJung @Centril (perhaps?) -- do we expect that these warnings will eventually migrate to errors? AFAIK, they're always UB, right?

Comment (Centril) [reply RELEASES.md]: I think this bit should explicitly state that these cases are UB.

> AFAIK, they're always UB, right?

Yep. The lint is a conservative underestimate of known UB.

File: RELEASES.md [hunk 5]
```diff
@@ -1,3 +1,113 @@
...
+
+Compiler
+--------
+- [Added support for the `aarch64-uwp-windows-msvc`, `i686-uwp-windows-gnu`,
+  `i686-uwp-windows-msvc`, `x86_64-uwp-windows-gnu`, and
+  `x86_64-uwp-windows-msvc` targets.][60260]
+- [Added support for the `armv7-unknown-linux-gnueabi` and
+  `armv7-unknown-linux-musleabi` targets.][63107]
+- [Added support for the `hexagon-unknown-linux-musl` target.][62814]
+- [Added pipelined compilation support to `rustc`.][62766] This will
```
Comment (BO41) [inline RELEASES.md]: maybe group all new targets into a single point.
Or at least move the pipeline feature to the beginning or the end

**Review by ghost — CHANGES_REQUESTED** [review]

File: RELEASES.md [hunk 6]
```diff
@@ -1,3 +1,113 @@
...
+- [`Duration::as_secs_f64`]
+- [`Duration::div_duration_f32`]
+- [`Duration::div_duration_f64`]
+- [`Duration::div_f32`]
+- [`Duration::div_f64`]
+- [`Duration::from_secs_f32`]
+- [`Duration::from_secs_f64`]
+- [`Duration::mul_f32`]
+- [`Duration::mul_f64`]
+- [`any::type_name`]
```
Comment (killercup) [inline RELEASES.md]: link missing, ref link name is <code>[\`intrinsics::type_name\`]</code>

File: RELEASES.md (hunk 4 above)
Comment (RalfJung) [reply RELEASES.md]: However, the phrasing "will warn if you create an invalid value" promises too much IMO.

"Will now detect some cases of invalid values" or so is better, I think. We want to clearly communicate that this is conservative and there are many many cases of invalid values that the lint will not catch.

File: RELEASES.md [hunk 7]
```diff
@@ -1,3 +1,113 @@
...
+Compiler
+--------
+- [Added support for the `aarch64-uwp-windows-msvc`, `i686-uwp-windows-gnu`,
+  `i686-uwp-windows-msvc`, `x86_64-uwp-windows-gnu`, and
+  `x86_64-uwp-windows-msvc` targets.][60260]
+- [Added support for the `armv7-unknown-linux-gnueabi` and
+  `armv7-unknown-linux-musleabi` targets.][63107]
+- [Added support for the `hexagon-unknown-linux-musl` target.][62814]
+- [Added pipelined compilation support to `rustc`.][62766] This will
+  improve compiliation times in some cases. For further information please refer
```
Comment (jonas-schievink) [inline RELEASES.md]:
```suggestion
  improve compilation times in some cases. For further information please refer
```

File: RELEASES.md (hunk 4 above)
Comment (XAMPPRocky) [reply RELEASES.md]: I've updated the wording, let me know if it matches what you expect.

Comment (RalfJung) [reply RELEASES.md]: Hm, that still sounds like we *are* catching all known cases of UB...

What about something like

> `rustc` will now warn about some incorrect uses of `mem::{uninitialized, zeroed}` that cause undefined behavior.

**Pull request merged**